src/java/com/puppetlabs/jruby_utils/pool/JRubyPool.java

```diff
@@ -192,15 +204,11 @@ public E borrowItemWithTimeout(long timeout, TimeUnit unit) throws
 
     @Override
     public void releaseItem(E e) {
-        releaseItem(e, true);
-    }
-
-    @Override
-    public void releaseItem(E e, boolean returnToPool) {
         final ReentrantLock lock = this.queueLock;
         lock.lock();
         try {
-            if (returnToPool) {
+            // Do nothing if the item being returned is the poisonPill
```

Comment:
> Another option here might be to see if the item being returned is something different than the poison pill or one of the previously registered pool instances and, if not, throw an IllegalArgumentException or something. Seems like returning something that was never in the pool in the first place should be considered invalid. That said, though, not sure if the extra validation is worth it. I think I'm good with this as it is. WDYT, @KevinCorcoran?

This seems like a decent approach.  The extra validation is probably unnecessary from a functionality perspective, but it seems like it would make the code a little easier to understand - i.e. Future Us won't have to look at this and wonder "what happens if it's a different pill?"